### DISCUSSION TITLE
unable to create a settlement report older using createReport Method.

### DISCUSSION THREAD

**Discussion by usman-rind** [amzn/selling-partner-api-models#338] [Q&A]
when i am calling create report Method to generate any report as i am getting below error. 

      "code": "InvalidInput",
      "message": "Request for report type 1118 is not allowed at this time",
      
      
      Below is my code. 
      
      `var list = new List<string>();
            list.Add("ATVPDKIKX0DER");

            // Define the date strings in the format "yyyy-MM-dd"
            string dateStr1 = "2023-05-03";
            string dateStr2 = "2023-05-17";

            // Parse the date strings into DateTime objects
            DateTime date1 = DateTime.ParseExact(dateStr1, "yyyy-MM-dd", CultureInfo.InvariantCulture);
            DateTime date2 = DateTime.ParseExact(dateStr2, "yyyy-MM-dd", CultureInfo.InvariantCulture);

            CreateReportSpecification specification = new CreateReportSpecification(
                null,
                reportType: "GET_V2_SETTLEMENT_REPORT_DATA_FLAT_FILE_V2",
                dataStartTime: date1, dataEndTime: date2,
                marketplaceIds: list
                );

           CreateReportResponse generatedReports = RApi.CreateReport(specification);`

**Comment by ritikacd** [comment]
Hi @usman-rind 

Settlement reports cannot be requested or scheduled. They are automatically scheduled by Amazon. You can search for these reports using the [getReports](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference#getreports) operation passing reportTypes parameter set to "SETTLEMENT" to check when the last settlement report was available.

Thanks,
Ritika Chaudhary
Selling Partner API Team
